<a href="https://colab.research.google.com/github/vunnampragathi/690-stats-and-visu/blob/main/Assignment4/assignment4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# assignment 4

name: Pragathi

date:09/22/2022

using 2017-2018 file from the college score card data

In [1]:
import pandas as pd
data = pd.read_csv("/content/MERGED2017_18_PP.csv")        # reading the 2017-18 data
data

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:3326: DtypeWarning: Columns (1729) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


UNITID    OPEID  OPEID6  \
0       100654   100200    1002   
1       100663   105200    1052   
2       100690  2503400   25034   
3       100706   105500    1055   
4       100724   100500    1005   
...        ...      ...     ...   
7107  48312404   108163    1081   
7108  48387801  4223701   42237   
7109  48400201  4228101   42281   
7110  48778201  4221501   42215   
7111  49005401  4182601   41826   

                                                 INSTNM             CITY  \
0                              Alabama A & M University           Normal   
1                   University of Alabama at Birmingham       Birmingham   
2                                    Amridge University       Montgomery   
3                   University of Alabama in Huntsville       Huntsville   
4                              Alabama State University       Montgomery   
...                                                 ...              ...   
7107                   Arizona State University at Yuma             Yuma   
7108  Bay Area Medical Academy - San Jose Satellite ...         San Jose   
7109           High Desert Medical College - Bakerfield      Bakersfield   
7110                                       BCI - Malden           Malden   
7111               HCI College - Fort Lauderdale Campus  Fort Lauderdale   

     STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
0        AL       35762           NaN      NaN     NaN  ...   
1        AL  35294-0110           NaN      NaN     NaN  ...   
2        AL  36117-3553           NaN      NaN     NaN  ...   
3        AL       35899           NaN      NaN     NaN  ...   
4        AL  36104-0271           NaN      NaN     NaN  ...   
...     ...         ...           ...      ...     ...  ...   
7107     AZ   853656900           NaN      NaN     NaN  ...   
7108     CA       95113           NaN      NaN     NaN  ...   
7109     CA       93301           NaN      NaN     NaN  ...   
7110     MA   021480000           NaN      NaN     NaN  ...   
7111     FL       33309           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN  \
0                              NaN       0.7106        0.7418   5207.0   
1                              NaN       0.4009        0.5488  13308.0   
2                              NaN       0.5000        0.5000    431.0   
3                              NaN       0.2012        0.4231   7519.0   
4                              NaN       0.7594        0.7402   5229.0   
...                            ...          ...           ...      ...   
7107                           NaN          NaN           NaN      NaN   
7108                           NaN          NaN           NaN      NaN   
7109                           NaN          NaN           NaN      NaN   
7110                           NaN          NaN           NaN      NaN   
7111                           NaN          NaN           NaN      NaN   

       G12MN  SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
0     1185.0   1410.0                 NaN                      NaN   
1     9888.0   1948.0                 NaN                      NaN   
2      437.0      4.0                 NaN                      NaN   
3     2378.0   1203.0                 NaN                      NaN   
4      687.0   1143.0                 NaN                      NaN   
...      ...      ...                 ...                      ...   
7107     NaN      NaN                 NaN                      NaN   
7108     NaN      NaN                 NaN                      NaN   
7109     NaN      NaN                 NaN                      NaN   
7110     NaN      NaN                 NaN                      NaN   
7111     NaN      NaN                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
0                          NaN             NaN  
1                          NaN             NaN  
2                          NaN          

In [2]:
data.shape   # displaying the size of the data which gives number of rows and columns

(7112, 1986)

number of rows or colleges is 7112

number of columns or variables is 1986

Looking at the first 5 colleges

In [3]:
data.head(5)   # printing first 5 rows in the data

UNITID    OPEID  OPEID6                               INSTNM        CITY  \
0  100654   100200    1002             Alabama A & M University      Normal   
1  100663   105200    1052  University of Alabama at Birmingham  Birmingham   
2  100690  2503400   25034                   Amridge University  Montgomery   
3  100706   105500    1055  University of Alabama in Huntsville  Huntsville   
4  100724   100500    1005             Alabama State University  Montgomery   

  STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
0     AL       35762           NaN      NaN     NaN  ...   
1     AL  35294-0110           NaN      NaN     NaN  ...   
2     AL  36117-3553           NaN      NaN     NaN  ...   
3     AL       35899           NaN      NaN     NaN  ...   
4     AL  36104-0271           NaN      NaN     NaN  ...   

   OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN   G12MN  \
0                           NaN       0.7106        0.7418   5207.0  1185.0   
1                           NaN       0.4009        0.5488  13308.0  9888.0   
2                           NaN       0.5000        0.5000    431.0   437.0   
3                           NaN       0.2012        0.4231   7519.0  2378.0   
4                           NaN       0.7594        0.7402   5229.0   687.0   

   SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
0   1410.0                 NaN                      NaN   
1   1948.0                 NaN                      NaN   
2      4.0                 NaN                      NaN   
3   1203.0                 NaN                      NaN   
4   1143.0                 NaN                      NaN   

   FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
0                       NaN             NaN  
1                       NaN             NaN  
2                       NaN             NaN  
3                       NaN             NaN  
4                       NaN             NaN  

[5 rows x 1986 columns]

Looking at the last 5 colleges

In [4]:
data.tail(5)  # printing last 5 rows in data

UNITID    OPEID  OPEID6  \
7107  48312404   108163    1081   
7108  48387801  4223701   42237   
7109  48400201  4228101   42281   
7110  48778201  4221501   42215   
7111  49005401  4182601   41826   

                                                 INSTNM             CITY  \
7107                   Arizona State University at Yuma             Yuma   
7108  Bay Area Medical Academy - San Jose Satellite ...         San Jose   
7109           High Desert Medical College - Bakerfield      Bakersfield   
7110                                       BCI - Malden           Malden   
7111               HCI College - Fort Lauderdale Campus  Fort Lauderdale   

     STABBR        ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
7107     AZ  853656900           NaN      NaN     NaN  ...   
7108     CA      95113           NaN      NaN     NaN  ...   
7109     CA      93301           NaN      NaN     NaN  ...   
7110     MA  021480000           NaN      NaN     NaN  ...   
7111     FL      33309           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN  UG12MN  G12MN  \
7107                           NaN          NaN           NaN     NaN    NaN   
7108                           NaN          NaN           NaN     NaN    NaN   
7109                           NaN          NaN           NaN     NaN    NaN   
7110                           NaN          NaN           NaN     NaN    NaN   
7111                           NaN          NaN           NaN     NaN    NaN   

      SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
7107      NaN                 NaN                      NaN   
7108      NaN                 NaN                      NaN   
7109      NaN                 NaN                      NaN   
7110      NaN                 NaN                      NaN   
7111      NaN                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
7107                       NaN             NaN  
7108                       NaN             NaN  
7109                       NaN             NaN  
7110                       NaN             NaN  
7111                       NaN             NaN  

[5 rows x 1986 columns]

Looking at a random sample of 5 colleges

In [5]:
data.sample(5)  # printing random 5 rows from the data

UNITID    OPEID  OPEID6                               INSTNM  \
3165  216454  2174400   21744             Triangle Tech Inc-Dubois   
6498  487913  4240600   42406             Tri-State Barber College   
1261  155353  2134800   21348      Old Town Barber College-Wichita   
2895  209108   319900    3199                Marylhurst University   
3428  224271  1076100   10761  Dallas Institute of Funeral Service   

             CITY STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
3165  Falls Creek     PA  15840-3333           NaN      NaN     NaN  ...   
6498      Ashland     KY  41101-7746           NaN      NaN     NaN  ...   
1261      Wichita     KS  67211-1605           NaN      NaN     NaN  ...   
2895   Marylhurst     OR  97036-0261           NaN      NaN     NaN  ...   
3428       Dallas     TX  75227-4314           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN  UG12MN  G12MN  \
3165                           NaN       0.5278        0.7500   246.0    NaN   
6498                           NaN       0.3333        0.3333    29.0    NaN   
1261                           NaN       0.8108        0.8649    80.0    NaN   
2895                           NaN          NaN           NaN   570.0  556.0   
3428                           NaN       0.5000        0.5000   204.0    NaN   

      SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
3165     72.0                 NaN                      NaN   
6498      6.0                 NaN                      NaN   
1261     37.0                 NaN                      NaN   
2895      0.0                 NaN                      NaN   
3428     24.0                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
3165                       NaN             NaN  
6498                       NaN             NaN  
1261                       NaN             NaN  
2895                       NaN             NaN  
3428                       NaN             NaN  

[5 rows x 1986 columns]

Since there are too many variables to explore, many of which are not interesting. Create a smaller data frame with the following 9 variables:

UNITID,INSTNM,CITY,STABBR,ZIP,ADM_RATE,UGDS,TUITIONFEE_IN,MN_EARN_WNE_P10

In [6]:
# creating a smaller data set with the required columns
new_data=data[['UNITID','INSTNM','CITY','STABBR','ZIP','ADM_RATE','UGDS','TUITIONFEE_IN','MN_EARN_WNE_P10']]
new_data

UNITID                                             INSTNM  \
0       100654                           Alabama A & M University   
1       100663                University of Alabama at Birmingham   
2       100690                                 Amridge University   
3       100706                University of Alabama in Huntsville   
4       100724                           Alabama State University   
...        ...                                                ...   
7107  48312404                   Arizona State University at Yuma   
7108  48387801  Bay Area Medical Academy - San Jose Satellite ...   
7109  48400201           High Desert Medical College - Bakerfield   
7110  48778201                                       BCI - Malden   
7111  49005401               HCI College - Fort Lauderdale Campus   

                 CITY STABBR         ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  \
0              Normal     AL       35762    0.9027   4824.0         9857.0   
1          Birmingham     AL  35294-0110    0.9181  12866.0         8328.0   
2          Montgomery     AL  36117-3553       NaN    318.0         6900.0   
3          Huntsville     AL       35899    0.8123   6917.0        10280.0   
4          Montgomery     AL  36104-0271    0.9787   4189.0        11068.0   
...               ...    ...         ...       ...      ...            ...   
7107             Yuma     AZ   853656900       NaN      NaN            NaN   
7108         San Jose     CA       95113       NaN      NaN            NaN   
7109      Bakersfield     CA       93301       NaN      NaN            NaN   
7110           Malden     MA   021480000       NaN      NaN            NaN   
7111  Fort Lauderdale     FL       33309       NaN      NaN        22575.0   

      MN_EARN_WNE_P10  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4                 NaN  
...               ...  
7107              NaN  
7108              NaN  
7109              NaN  
7110              NaN  
7111              NaN  

[7112 rows x 9 columns]

**UNITID**:Integer

Data files are provided at the UNITID level, which is the unique
identification number assigned to postsecondary institutions as
surveyed through IPEDS.

**INSTNM**:String

The institution’s name (INSTNM), as reported in IPEDS.

***CITY:***String

The institution’s location, as reported in IPEDS 

**STABR**:String

The institution’s location, as reported in IPEDS 

**ZIP**:String
The institution’s location, as reported in IPEDS 

**ADM_RATE**:Float

Colleges report to IPEDS their Fall admissions rate, defined as the
number of admitted undergraduates divided by the number of
undergraduates who applied. For institutions with multiple branches,
ADM_RATE includes the admissions rate at each campus, while
ADM_RATE_ALL represents the admissions rate across all campuses,
defined as the total number of admitted undergraduates across all
branches divided by the total number of undergraduates who applied
across all branches. Please note these are also often referred to as
acceptance rates


**UGDS**: Integer

This element (UGDS) includes the number of degree/certificate-seeking
undergraduates enrolled in the fall, as reported in the IPEDS Fall
Enrollment component

**TUITIONFEE_IN**:Tuition
and required fees are provided for in-state students (TUITIONFEE_IN)

**MN_EARN_WNE_P10**:Mean (MN_EARN_WNE_P*) earnings are for the institutional
aggregate of all federally aided undergraduate students who
enroll in an institution each year and who are employed but not
enrolled

Look at the first 10 colleges

In [7]:
new_data.head(10)   # printing first 10 rows of new data

UNITID                               INSTNM            CITY STABBR  \
0  100654             Alabama A & M University          Normal     AL   
1  100663  University of Alabama at Birmingham      Birmingham     AL   
2  100690                   Amridge University      Montgomery     AL   
3  100706  University of Alabama in Huntsville      Huntsville     AL   
4  100724             Alabama State University      Montgomery     AL   
5  100751            The University of Alabama      Tuscaloosa     AL   
6  100760    Central Alabama Community College  Alexander City     AL   
7  100812              Athens State University          Athens     AL   
8  100830      Auburn University at Montgomery      Montgomery     AL   
9  100858                    Auburn University          Auburn     AL   

          ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
0       35762    0.9027   4824.0         9857.0              NaN  
1  35294-0110    0.9181  12866.0         8328.0              NaN  
2  36117-3553       NaN    318.0         6900.0              NaN  
3       35899    0.8123   6917.0        10280.0              NaN  
4  36104-0271    0.9787   4189.0        11068.0              NaN  
5  35487-0100    0.5330  32387.0        10780.0              NaN  
6       35010       NaN   1404.0         4440.0              NaN  
7       35611       NaN   2801.0            NaN              NaN  
8  36117-3596    0.8240   4211.0         8020.0              NaN  
9       36849    0.8393  23391.0        10968.0              NaN

Look at the last 10 colleges

In [8]:
new_data.tail(10)  # printing last 10 rows of the new data

UNITID                                             INSTNM  \
7102  48154401  National Personal Training Institute of Cleveland   
7103  48305401             Barber School of Pittsburgh - Ambridge   
7104  48312401                 Arizona State University at Tucson   
7105  48312402  Arizona State University - ASU Colleges at Lak...   
7106  48312403        Arizona State University at The Gila Valley   
7107  48312404                   Arizona State University at Yuma   
7108  48387801  Bay Area Medical Academy - San Jose Satellite ...   
7109  48400201           High Desert Medical College - Bakerfield   
7110  48778201                                       BCI - Malden   
7111  49005401               HCI College - Fort Lauderdale Campus   

                  CITY STABBR        ZIP  ADM_RATE  UGDS  TUITIONFEE_IN  \
7102  Highland Heights     OH      44143       NaN   NaN            NaN   
7103          Ambridge     PA  150030000       NaN   NaN            NaN   
7104            Tucson     AZ  857454284       NaN   NaN            NaN   
7105  Lake Havasu City     AZ  864036877       NaN   NaN         6904.0   
7106          Thatcher     AZ  855525545       NaN   NaN            NaN   
7107              Yuma     AZ  853656900       NaN   NaN            NaN   
7108          San Jose     CA      95113       NaN   NaN            NaN   
7109       Bakersfield     CA      93301       NaN   NaN            NaN   
7110            Malden     MA  021480000       NaN   NaN            NaN   
7111   Fort Lauderdale     FL      33309       NaN   NaN        22575.0   

      MN_EARN_WNE_P10  
7102              NaN  
7103              NaN  
7104              NaN  
7105              NaN  
7106              NaN  
7107              NaN  
7108              NaN  
7109              NaN  
7110              NaN  
7111              NaN

Look at the random sample of 10 colleges

In [9]:
new_data.sample(10)   # printing random 10 rows of new data

UNITID                                             INSTNM  \
334   116350       Toni & Guy Hairdressing Academy-Santa Monica   
3727  234933                                      Clark College   
768   137591                              Bene's Career Academy   
3475  226374                               Lubbock Hair Academy   
1595  167288              Massachusetts College of Liberal Arts   
198   109819                                Bakersfield College   
739   136330                     Palm Beach Atlantic University   
2620  200624  The Art Institute of Cincinnati-AIC College of...   
1513  164580                                     Babson College   
5124  446039  Arizona School of Acupuncture and Oriental Med...   

                 CITY STABBR         ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  \
334      Santa Monica     CA       90401       NaN    222.0            NaN   
3727        Vancouver     WA  98663-3598       NaN   8170.0         3613.0   
768   New Port Richey     FL       34652       NaN    305.0            NaN   
3475          Lubbock     TX       79410       NaN     45.0            NaN   
1595      North Adams     MA  01247-4100    0.7663   1364.0        10135.0   
198       Bakersfield     CA  93305-1299       NaN  18695.0         1325.0   
739   West Palm Beach     FL       33401    0.9739   2475.0        29950.0   
2620       Cincinnati     OH  45246-3322    0.4091     20.0        23341.0   
1513        Wellesley     MA  02457-0310    0.2418   2342.0        49664.0   
5124           Tucson     AZ  85716-1518       NaN      NaN            NaN   

      MN_EARN_WNE_P10  
334               NaN  
3727              NaN  
768               NaN  
3475              NaN  
1595              NaN  
198               NaN  
739               NaN  
2620              NaN  
1513              NaN  
5124              NaN

Find and display UMBC

In [10]:
df1=new_data.loc[new_data['INSTNM'] == "University of Maryland-Baltimore County"]  # searching for umbc
df1

UNITID                                   INSTNM       CITY STABBR  \
1478  163268  University of Maryland-Baltimore County  Baltimore     MD   

        ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1478  21250    0.6021  11130.0        11518.0              NaN

Find Univ. of Maryland College Park

In [11]:
df2=new_data.loc[new_data['INSTNM'] == "University of Maryland-College Park"] # searching for college park
df2

UNITID                               INSTNM          CITY STABBR    ZIP  \
1479  163286  University of Maryland-College Park  College Park     MD  20742   

      ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1479     0.445  29273.0        10399.0              NaN

Find and display Loyola College Maryland

In [12]:
df3=new_data.loc[new_data['INSTNM'] == "Loyola University Maryland"]   #searching for loyola college
df3

UNITID                      INSTNM       CITY STABBR         ZIP  \
1475  163046  Loyola University Maryland  Baltimore     MD  21210-2699   

      ADM_RATE    UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1475    0.7528  3903.0        47560.0              NaN

Find and display Johns Hopskins


In [13]:
df4=new_data.loc[new_data['INSTNM'] == "Johns Hopkins University"]  # searching for john hopkins
df4

UNITID                    INSTNM       CITY STABBR         ZIP  \
1473  162928  Johns Hopkins University  Baltimore     MD  21218-2688   

      ADM_RATE    UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1473    0.1254  5663.0        52170.0              NaN

In [14]:
df = pd.concat([df1,df2, df3,df4], ignore_index=True)       # joining all the above dataframes to form small new dataframe using concat function
df

UNITID                                   INSTNM          CITY STABBR  \
0  163268  University of Maryland-Baltimore County     Baltimore     MD   
1  163286      University of Maryland-College Park  College Park     MD   
2  163046               Loyola University Maryland     Baltimore     MD   
3  162928                 Johns Hopkins University     Baltimore     MD   

          ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
0       21250    0.6021  11130.0        11518.0              NaN  
1       20742    0.4450  29273.0        10399.0              NaN  
2  21210-2699    0.7528   3903.0        47560.0              NaN  
3  21218-2688    0.1254   5663.0        52170.0              NaN

Calculate the min, max, and average in-state tution of the above four institutions.

In [15]:
df[df["TUITIONFEE_IN"] == df["TUITIONFEE_IN"].max()]  # finidng the row with maximum tutuion fee

UNITID                    INSTNM       CITY STABBR         ZIP  ADM_RATE  \
3  162928  Johns Hopkins University  Baltimore     MD  21218-2688    0.1254   

     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
3  5663.0        52170.0              NaN

In [16]:
df[df["TUITIONFEE_IN"] == df["TUITIONFEE_IN"].min()]   # finding the row with minimum tutuion fee

UNITID                               INSTNM          CITY STABBR    ZIP  \
1  163286  University of Maryland-College Park  College Park     MD  20742   

   ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1     0.445  29273.0        10399.0              NaN

In [17]:
df["TUITIONFEE_IN"].mean()    # finding the average tution fee of all the four universites

30411.75

Filter the smaller dataset further to only keep colleges from Maryland, Virgina, Washington DC, Delaware, and Pennsylvinia. Save the filtered dataset to a new file named "Five_States_Colleges.csv" for future analysis

In [18]:
new_df1=new_data.loc[new_data['STABBR'] == "MD"]    # searching for colleges in maryland
new_df1

UNITID                                  INSTNM          CITY STABBR  \
1446    161651     Aesthetics Institute of Cosmetology  Gaithersburg     MD   
1447    161688            Allegany College of Maryland    Cumberland     MD   
1448    161767          Anne Arundel Community College        Arnold     MD   
1449    161776               Brightwood College-Towson        Towson     MD   
1450    161800                     Award Beauty School    Hagerstown     MD   
...        ...                                     ...           ...    ...   
6800  16210401          Cecil College - Elkton Station        Elkton     MD   
7004  43018401     Strayer University-Rockville Campus     Rockville     MD   
7005  43018402  Strayer University-Anne Arundel Campus  Millersville     MD   
7006  43018403   Strayer University-White Marsh Campus     Baltimore     MD   
7007  43018404  Strayer University-Owings Mills Campus  Owings Mills     MD   

             ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1446       20877       NaN     25.0            NaN              NaN  
1447  21502-2596       NaN   1990.0         3940.0              NaN  
1448  21012-1895       NaN  11294.0         4646.0              NaN  
1449  21286-2201       NaN    309.0            NaN              NaN  
1450  21740-5610       NaN    157.0            NaN              NaN  
...          ...       ...      ...            ...              ...  
6800   219215535       NaN      NaN         4005.0              NaN  
7004       20850       NaN      NaN        13857.0              NaN  
7005       21108       NaN      NaN        13857.0              NaN  
7006       21236       NaN      NaN        13857.0              NaN  
7007       21117       NaN      NaN        13857.0              NaN  

[91 rows x 9 columns]

In [19]:
new_df2=new_data.loc[new_data['STABBR'] == "DC"]    # searching for colleges in washington DC
new_df2

UNITID                                             INSTNM        CITY  \
643     131159                                American University  Washington   
644     131283                 The Catholic University of America  Washington   
645     131399             University of the District of Columbia  Washington   
646     131405  Pontifical Faculty of the Immaculate Conceptio...  Washington   
647     131450                               Gallaudet University  Washington   
648     131469                       George Washington University  Washington   
649     131496                              Georgetown University  Washington   
650     131520                                  Howard University  Washington   
651     131803            Strayer University-District of Columbia  Washington   
652     131830             National Conservatory of Dramatic Arts  Washington   
653     131876                      Trinity Washington University  Washington   
654     131973                        Wesley Theological Seminary  Washington   
4100    363721  University of the District of Columbia-David A...  Washington   
4348    384412     University of the Potomac-Washington DC Campus  Washington   
4616    420370                         Career Technical Institute  Washington   
4844    439473                           Bennett Career Institute  Washington   
5079    444893                     Technical Learning Centers Inc  Washington   
5328    450483                University of Phoenix-Washington DC  Washington   
5520    455804                        Institute of World Politics  Washington   
5521    455813  Pontifical John Paul II Institute for Studies ...  Washington   
5739    459745  The Chicago School of Professional Psychology ...  Washington   
5747    459994                   Strayer University-Global Region  Washington   
6727  13180301              Strayer University-Takoma Park Campus  Washington   
7094  47503313  Relay Graduate School of Education - Washingto...  Washington   

     STABBR         ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
643      DC  20016-8001    0.2940   7433.0        46615.0              NaN  
644      DC       20064    0.8258   3284.0        44060.0              NaN  
645      DC       20008       NaN   3606.0         5756.0              NaN  
646      DC  20017-1585       NaN      NaN            NaN              NaN  
647      DC  20002-3695    0.5901   1111.0        16558.0              NaN  
648      DC       20052    0.4098  11718.0        53518.0              NaN  
649      DC  20057-0001    0.1568   7128.0        52300.0              NaN  
650      DC  20059-0001    0.4142   6276.0        25697.0              NaN  
651      DC       20005       NaN    687.0        13857.0              NaN  
652      DC       20007    1.0000     32.0         9900.0              NaN  
653      DC  20017-1094    0.9228   1498.0        23690.0              NaN  
654      DC  20016-5690       NaN      NaN            NaN              NaN  
4100     DC       20008       NaN      NaN            NaN              NaN  
4348     DC       20005       NaN     65.0        13884.0              NaN  
4616     DC       20005       NaN    387.0            NaN              NaN  
4844     DC       20017       NaN    312.0            NaN              NaN  
5079     DC  20006-5540       NaN    172.0            NaN              NaN  
5328     DC  20001-1431       NaN     70.0         9960.0              NaN  
5520     DC       20036       NaN      NaN            NaN              NaN  
5521     DC       20064       NaN      NaN            NaN              NaN  
5739     DC       20005       NaN      NaN            NaN              NaN  
5747     DC       20005       NaN   5543.0        13857.0              NaN  
6727     DC       20012       NaN      NaN        13857.0              NaN  
7094     DC       20002       NaN      NaN            NaN              NaN

In [20]:
new_df3=new_data.loc[new_data['STABBR'] == "VA"]   # searcing for colleges in viriginia
new_df3

UNITID                                          INSTNM  \
3627    231280                Paul Mitchell the School-Roanoke   
3628    231411                   Advanced Technology Institute   
3629    231420                              Averett University   
3630    231536                    Blue Ridge Community College   
3631    231554                               Bluefield College   
...        ...                                             ...   
6969  24893411  ECPI University-Virginia Beach Health Sciences   
6970  24893412                  ECPI University-Richmond South   
6971  24893413                       ECPI University-Innsbrook   
6972  24893414                   ECPI University-Richmond West   
6973  24893415                       ECPI University - Roanoke   

                CITY STABBR         ZIP  ADM_RATE    UGDS  TUITIONFEE_IN  \
3627         Roanoke     VA       24015       NaN    84.0            NaN   
3628  Virginia Beach     VA       23462    0.7954   464.0        13250.0   
3629        Danville     VA       24541    0.6102   908.0        33350.0   
3630     Weyers Cave     VA  24486-0080       NaN  3054.0         5252.0   
3631       Bluefield     VA       24605    0.9112   874.0        24800.0   
...              ...    ...         ...       ...     ...            ...   
6969  Virginia Beach     VA   234621603       NaN     NaN        15261.0   
6970        Richmond     VA   232361603       NaN     NaN        14985.0   
6971      Glen Allen     VA   230601603       NaN     NaN        15263.0   
6972        Richmond     VA   232941603       NaN     NaN        16039.0   
6973         Roanoke     VA   240121603       NaN     NaN        15525.0   

      MN_EARN_WNE_P10  
3627              NaN  
3628              NaN  
3629              NaN  
3630              NaN  
3631              NaN  
...               ...  
6969              NaN  
6970              NaN  
6971              NaN  
6972              NaN  
6973              NaN  

[177 rows x 9 columns]

In [21]:
new_df4=new_data.loc[new_data['STABBR'] == "PA"]    # searching for colleges in penselvinia
new_df4

UNITID                                             INSTNM  \
2933    210474             Jolie Hair and Beauty Academy-Hazleton   
2934    210492               Bryn Athyn College of the New Church   
2935    210508                              Academy of Vocal Arts   
2936    210571                                   Albright College   
2937    210599                            All-State Career School   
...        ...                                                ...   
7023  44378405               Strayer University-Warrendale Campus   
7024  44378406                Strayer University-Allentown Campus   
7063  45794301   Somerset County Technology Center - Bedford Site   
7089  47503308  Relay Graduate School of Education - Philadelp...   
7103  48305401             Barber School of Pittsburgh - Ambridge   

               CITY STABBR         ZIP  ADM_RATE    UGDS  TUITIONFEE_IN  \
2933       Hazleton     PA       18201       NaN    61.0            NaN   
2934     Bryn Athyn     PA  19009-0711    0.8755   323.0        21126.0   
2935   Philadelphia     PA  19103-6685       NaN     NaN            NaN   
2936        Reading     PA  19612-5234    0.5004  1999.0        43454.0   
2937      Essington     PA       19029       NaN   306.0            NaN   
...             ...    ...         ...       ...     ...            ...   
7023     Warrendale     PA       15086       NaN     NaN        13857.0   
7024  Center Valley     PA       18034       NaN     NaN        13857.0   
7063        Bedford     PA       15522       NaN     NaN            NaN   
7089   Philadelphia     PA   191224906       NaN     NaN            NaN   
7103       Ambridge     PA   150030000       NaN     NaN            NaN   

      MN_EARN_WNE_P10  
2933              NaN  
2934              NaN  
2935              NaN  
2936              NaN  
2937              NaN  
...               ...  
7023              NaN  
7024              NaN  
7063              NaN  
7089              NaN  
7103              NaN  

[378 rows x 9 columns]

In [22]:
new_df5=new_data.loc[new_data['STABBR'] == "DE"]   # searching for collegs in delaware
new_df5

UNITID                                             INSTNM        CITY  \
634     130837  Margaret H Rollins School of Nursing at Beebe ...       Lewes   
635     130873                          Dawn Career Institute LLC      Newark   
636     130907         Delaware Technical Community College-Terry       Dover   
637     130934                          Delaware State University       Dover   
638     130943                             University of Delaware      Newark   
639     130989                              Goldey-Beacom College  Wilmington   
640     131061        Schilling-Douglas School of Hair Design LLC      Newark   
641     131098                                     Wesley College       Dover   
642     131113                              Wilmington University  New Castle   
4729    432524                 Delaware College of Art and Design  Wilmington   
4917    441362        Harris School of Business-Wilmington Campus  Wilmington   
5037    444200             Harris School of Business-Dover Campus       Dover   
5275    449472         Delaware Learning Institute of Cosmetology    Dagsboro   
5322    450298                        Strayer University-Delaware      Newark   
5354    451051                    Academy of Massage and Bodywork        Bear   
5718    459301                  Paul Mitchell the School-Delaware      Newark   
6527    488262          Hair Academy School of Barbering & Beauty      Newark   
6725  13090701         Delaware Technical Community College-Owens  Georgetown   
6726  13090702  Delaware Technical Community College-Stanton/W...  Wilmington   
7084  47503303      Relay Graduate School of Education - Delaware  Wilmington   

     STABBR         ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
634      DE       19958       NaN     59.0            NaN              NaN  
635      DE       19702       NaN    108.0            NaN              NaN  
636      DE       19901       NaN  12408.0         4848.0              NaN  
637      DE       19901    0.4518   3678.0         7868.0              NaN  
638      DE       19716    0.6289  18948.0        13160.0              NaN  
639      DE       19808    0.5177    770.0        23850.0              NaN  
640      DE       19711       NaN    133.0            NaN              NaN  
641      DE  19901-3875    0.6151   1342.0        26406.0              NaN  
642      DE       19720       NaN   8346.0        10940.0              NaN  
4729     DE  19801-3007    0.7900    143.0        25710.0              NaN  
4917     DE       19803       NaN    219.0            NaN              NaN  
5037     DE       19901       NaN    226.0            NaN              NaN  
5275     DE       19939       NaN     39.0            NaN              NaN  
5322     DE       19713       NaN    267.0        13857.0              NaN  
5354     DE       19701       NaN     95.0            NaN              NaN  
5718     DE       19702       NaN    103.0            NaN              NaN  
6527     DE  19713-2305       NaN     33.0            NaN              NaN  
6725     DE       19947       NaN      NaN         4848.0              NaN  
6726     DE       19801       NaN      NaN         4848.0              NaN  
7084     DE       19801       NaN      NaN            NaN              NaN

In [23]:
final_df = pd.concat([new_df1,new_df2,new_df3,new_df4,new_df5], ignore_index=True)   # concatinating all the above 5 dataframes to get new dataframe
final_df

UNITID                                             INSTNM  \
0      161651                Aesthetics Institute of Cosmetology   
1      161688                       Allegany College of Maryland   
2      161767                     Anne Arundel Community College   
3      161776                          Brightwood College-Towson   
4      161800                                Award Beauty School   
..        ...                                                ...   
685    459301                  Paul Mitchell the School-Delaware   
686    488262          Hair Academy School of Barbering & Beauty   
687  13090701         Delaware Technical Community College-Owens   
688  13090702  Delaware Technical Community College-Stanton/W...   
689  47503303      Relay Graduate School of Education - Delaware   

             CITY STABBR         ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  \
0    Gaithersburg     MD       20877       NaN     25.0            NaN   
1      Cumberland     MD  21502-2596       NaN   1990.0         3940.0   
2          Arnold     MD  21012-1895       NaN  11294.0         4646.0   
3          Towson     MD  21286-2201       NaN    309.0            NaN   
4      Hagerstown     MD  21740-5610       NaN    157.0            NaN   
..            ...    ...         ...       ...      ...            ...   
685        Newark     DE       19702       NaN    103.0            NaN   
686        Newark     DE  19713-2305       NaN     33.0            NaN   
687    Georgetown     DE       19947       NaN      NaN         4848.0   
688    Wilmington     DE       19801       NaN      NaN         4848.0   
689    Wilmington     DE       19801       NaN      NaN            NaN   

     MN_EARN_WNE_P10  
0                NaN  
1                NaN  
2                NaN  
3                NaN  
4                NaN  
..               ...  
685              NaN  
686              NaN  
687              NaN  
688              NaN  
689              NaN  

[690 rows x 9 columns]

In [24]:
    final_df.to_csv('Five_States_Colleges.csv', index=False)     # downloading the merged dataset with colleges data on 5 staes

ref:

 1.https://www.statology.org/pandas-select-rows-based-on-column-values/

2.https://www.statology.org/pandas-append-multiple-dataframes/